src/main/java/bio/terra/service/dataset/flight/ingest/IngestParseJsonFileStep.java

```diff
@@ -0,0 +1,101 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.common.Column;
+import bio.terra.model.BulkLoadFileModel;
+import bio.terra.model.IngestRequestModel;
+import bio.terra.model.TableDataType;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.exception.IngestFailureException;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IngestParseJsonFileStep implements Step {
+
+  private final GcsPdao gcsPdao;
+  private final Dataset dataset;
+  private final ObjectMapper objectMapper;
+
+  public IngestParseJsonFileStep(GcsPdao gcsPdao, ObjectMapper objectMapper, Dataset dataset) {
+    this.gcsPdao = gcsPdao;
+    this.objectMapper = objectMapper;
+    this.dataset = dataset;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    IngestRequestModel ingestRequest = IngestUtils.getIngestRequestModel(flightContext);
+    List<String> gcsFileLines =
+        gcsPdao.getGcsFilesLines(
+            ingestRequest.getPath(), dataset.getProjectResource().getGoogleProjectId());
+    List<String> fileRefColumnNames =
+        dataset.getTableByName(ingestRequest.getTable()).orElseThrow().getColumns().stream()
+            .filter(c -> c.getType() == TableDataType.FILEREF)
+            .map(Column::getName)
+            .collect(Collectors.toList());
+    var workingMap = flightContext.getWorkingMap();
+    workingMap.put(IngestMapKeys.TABLE_SCHEMA_FILE_COLUMNS, fileRefColumnNames);
+
+    List<String> errors = new ArrayList<>();
+    List<JsonNode> fileLineJson =
+        gcsFileLines.stream()
+            .map(
+                content -> {
+                  try {
+                    return objectMapper.readTree(content);
+                  } catch (JsonProcessingException ex) {
+                    errors.add(ex.getMessage());
+                    return null;
+                  }
+                })
+            .collect(Collectors.toList());
+
+    if (!errors.isEmpty()) {
+      IngestFailureException ex =
+          new IngestFailureException(
+              "Ingest control file at " + ingestRequest.getPath() + " could not be processed",
+              errors);
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, ex);
+    }
+
+    workingMap.put(IngestMapKeys.BULK_LOAD_JSON_LINES, fileLineJson);
```

Comment:
Is this storing the entire json load lines into the flight map? I thought these load files could be very large, is there no way to do this without storing all the data?

What if it only stored the path data as a Map<String, String> of original and uploaded path values?

Reply:
I think we don't necessarily need to put the entire json into the flight map, but we would then have to parse the ingest file a second time. We need the parsed json in the `IngestBuildLoadFileStep` so that we can replace in the file ingest requests w/ the new fileref and present the new json to BigQuery. 

I don't totally understand your suggested alternative. I _think_ we're doing this via the map created in `IngestBuildLoadFileStep` that maps the paths to the new fileid (see line 42). But, we still need the raw JSON. Maybe we can discuss in parking lot?

Reply:
Trevyn's idea: We could store a serialized version of the parsed json into firestore/azure storage tables. But, this is a larger change that would need more vetting around the team. 

We're going to limit the initial implementation of this feature to a small number of rows. So, it may be okay to keep the existing workflow for now.

Reply:
Per discussion, will make follow-on ticket.